src/GraphQL/Response/ViolationResponse.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\graphql\GraphQL\Response;
+
+use Drupal\graphql\GraphQL\Violation\ViolationCollection;
+
+/**
+ * Base class for responses containing the violations.
+ */
+class ViolationResponse implements ResponseInterface {
```

Comment:
It is still a bit weird to me that in the success case you still get a ViolationResponse although nothing went wrong. should this be "class MutationResponse"? If there is something in errors() then the mutation did not work.

Reply:
> should we combine this with ResponseInterface? Then we don't need this extra class. A data producer can just add violations on the mutation response object itself.

as I addressed that, it's maybe this is not relevant anymore?